docs/asciidoc/metrics.adoc

```diff
@@ -0,0 +1,127 @@
+====
+[source,java]
+----
+Schedulers.enableMetrics();
+----
+====
+
+WARNING: The instrumentation is performed when a scheduler is created. It is recommended to call this method as easly as possible.
+
+TIP: If you're using Spring Boot, it is a good idea to place the invocation before `SpringApplication.run(Application.class, args)` call.
+
+After you enable the scheduler metrics, Reactor will instrument (with Micrometer's built-in instrumentation) the underlying executors.
+
+Please refer to http://micrometer.io/docs[Micrometer's documentation] for the exposed metrics, such as:
+
+- executor_active_threads
+- executor_completed_tasks_total
+- executor_pool_size_threads
+- executor_queued_tasks
+- executor_secounds_{count, max, sum}
+
+Since one scheduler may have multiple executors, every executor metric has a `reactor_scheduler_id` tag.
+
+TIP: Grafana + Prometheus users can use https://raw.githubusercontent.com/reactor/reactor-monitoring-demo/master/dashboards/schedulers.json[a pre-built dashboard] which includes panels for threads, completed tasks, task queues and other handy metrics:
+
+== Publisher metrics
+Sometimes it is useful to be able to record metrics at some stage in your reactive pipeline.
+
+One way to do it would be to manually push the values to your metrics backend of choice.
+Another option would be to use Reactor's built-in metrics integration for `Flux`/`Mono` and interpret them.
+
+Consider the following pipeline:
+====
+[source,java]
+----
+listenToEvents()
+    .doOnNext(event -> log.info("Received {}", event))
+    .delayUntil(this::processEvent)
+    .retry()
+    .subscribe();
+----
+====
+
+To enable the metrics for this source `Flux` (returned from `listenToEvents()`), we need to give it a name and turn on the metrics collecting:
+====
+[source,java]
+----
+listenToEvents()
+    .name("events") <1>
+    .metrics() <2>
+    .doOnNext(event -> log.info("Received {}", event))
+    .delayUntil(this::processEvent)
+    .retry()
+    .subscribe();
+----
+<1> Every metric at this stage will be identified as "events".
+<2> `Flux#metrics` operator enables the reporting of metrics and uses the last known name up in the pipeline.
+====
+
+Just adding these two operators will expose a whole bunch of useful metrics!
+
+[width="100%",options="header"]
+|=======
+| metric name | type | description
+
+| reactor.subscribed | Counter | Counts how many Reactor sequences have been subscribed to
+
+| reactor.malformed.source | Counter | Counts the number of events received from a malformed source (ie an onNext after an onComplete)
+
+| reactor.requested | DistributionSummary | Counts the amount requested to a named Flux by all subscribers, until at least one requests an unbounded amount
+
+| reactor.onNext.delay | Timer | Measures delays between onNext signals (or between onSubscribe and first onNext)
+
+| reactor.flow.duration | Timer | Times the duration elapsed between a subscription and the cancellation of the sequence
```

Comment:
it is not only cancellation, but cancellation or termination (onComplete and onError). should mention the `status` and its 3 possible values.

Reply:
copied from the javadoc. Should I change it there too?

Reply:
yes, it should probably be changed in the javadoc. I'd suggest:
> Times the duration elapsed between a subscription and the termination or cancellation of the sequence. A `status` tag is added to specify what event caused the timer to end (`onComplete`, `onError`, `cancel`)